server/lib/realtime/metrics/socket_monitor.ex

```diff
@@ -0,0 +1,81 @@
+# This file draws from https://github.com/pushex-project/pushex
+# License: https://github.com/pushex-project/pushex/blob/master/LICENSE
+
+defmodule Realtime.Metrics.SocketMonitor do
+  use GenServer
+  alias Realtime.Metrics.PromEx.Plugins.Realtime
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def init(_) do
+    {:ok, %{transport_pids: %{}, channel_pids: %{}}}
+  end
+
+  def track_socket(socket = %Phoenix.Socket{}) do
+    GenServer.call(__MODULE__, {:track_socket, socket})
```

Comment:
I think it's better to use non-blocking callings `(handle_cast/handle_info)` because metrics can slow down accepting new sessions, the same for `track_channel`

Reply:
yes, agreed!